var/spack/repos/builtin/packages/tensorflow/package.py

```diff
@@ -0,0 +1,290 @@
+                        'tensorflow/tools/pip_package/setup.py')
+        if self.spec.satisfies('@1.8.0:'):
+            # 1.8.0 and 1.9.0 aborts with numpy import error during python_api
+            # generation somehow the wrong PYTHONPATH is used...
+            # set --distinct_host_configuration=false as a workaround
+            # (https://github.com/tensorflow/tensorflow/issues/22395#issuecomment-431229451)  # noqa: E501
+            filter_file('build --action_env TF_NEED_OPENCL_SYCL="0"',
+                        'build --action_env TF_NEED_OPENCL_SYCL="0"\n'
+                        'build --distinct_host_configuration=false\n'
+                        'build --action_env PYTHONPATH="{0}"'.format(env['PYTHONPATH']),  # noqa: E501
+                        '.tf_configure.bazelrc')
+        if self.spec.satisfies('@1.13.1'):
+            # tensorflow_estimator is an API for tensorflow
+            # tensorflow-estimator imports tensorflow during build, so
+            # tensorflow has to be set up first
+            filter_file(r"'tensorflow_estimator >=",
+                        r"#'tensorflow_estimator >=",
+                        'tensorflow/tools/pip_package/setup.py')
+        if self.spec.satisfies('@2.0.0:'):
+            # now it depends on the nightly versions...
+            filter_file(r"'tf-estimator-nightly >=",
+                        r"#'tf-estimator-nightly >=",
+                        'tensorflow/tools/pip_package/setup.py')
+            filter_file(r"REQUIRED_PACKAGES\[i\] = 'tb-nightly >=",
+                        r"pass #REQUIRED_PACKAGES\[i\] = 'tb-nightly >=",
+                        'tensorflow/tools/pip_package/setup.py')
+            filter_file(r"'tb-nightly >=",
+                        r"#'tb-nightly >=",
+                        'tensorflow/tools/pip_package/setup.py')
+
+        if self.spec.satisfies('+nccl'):
+            filter_file(r'^build --action_env NCCL_INSTALL_PATH=.*',
+                     r'build --action_env NCCL_INSTALL_PATH="'+str(spec['nccl'].prefix.lib)+'"',
+                     '.tf_configure.bazelrc')
+            filter_file(r'^build --action_env NCCL_HDR_PATH=.*',
+                     r'build --action_env NCCL_HDR_PATH="'+str(spec['nccl'].prefix.include)+'"',
+                    '.tf_configure.bazelrc')
+
+        if self.spec.satisfies('+cuda'):
+            libs = [
+                    str(spec['cuda'].prefix)+'/lib',
+                    str(spec['cuda'].prefix)+'/lib64',
+                    str(spec['cudnn'].prefix)+'/lib',
+                    str(spec['cudnn'].prefix)+'/lib64',
+                    str(spec['nccl'].prefix)+'/lib',
+                    str(spec['nccl'].prefix)+'/lib64']
+            slibs = ':'.join(libs)
+
+            filter_file('build --action_env TF_NEED_OPENCL_SYCL="0"',
+                        'build --action_env TF_NEED_OPENCL_SYCL="0"\n'
+                        'build --action_env LD_LIBRARY_PATH="'+slibs+'"',
```

Comment:
This is probably something that we want to take care on a bazel-level, however I'm not sure if we can... (i.e. not enough bazel ~~pain~~ experience)